apps/console/src/features/identity-providers/components/forms/helpers/form-fields-helper.tsx

```diff
@@ -316,9 +317,48 @@ export const getCopyInputField = (eachProp: CommonPluggableComponentPropertyInte
     );
 };
 
+export const getScopesField = (eachProp: CommonPluggableComponentPropertyInterface,
+    propertyMetadata: CommonPluggableComponentMetaPropertyInterface,
+    testId?: string): ReactElement => {
+
+    return (
+        <>
+            <Field
+                name={ propertyMetadata?.key }
+                label={ propertyMetadata?.displayName }
+                required={ propertyMetadata?.isMandatory }
+                requiredErrorMessage={ I18n.instance.t("console:develop.features.authenticationProvider.forms.common." +
+                    "requiredErrorMessage") }
+                type="scopes"
+                value={ eachProp?.value }
+                key={ eachProp?.key }
+                defaultValue={ propertyMetadata?.defaultValue }
+                disabled={ propertyMetadata?.isDisabled }
+                readOnly={ propertyMetadata?.readOnly }
+                data-testid={ `${ testId }-${ propertyMetadata?.key }` }
+                validation={ (value: string, validation: Validation) => {
+                    if (!FormValidation.scopes(value)) {
+                        validation.isValid = false;
+                        validation.errorMessages.push(I18n.instance.t("console:develop.features." +
+                            "authenticationProvider.forms.common.invalidScopesErrorMessage"));
+                    }
+                    if (propertyMetadata?.maxLength && value.length > propertyMetadata.maxLength) {
+                        validation.isValid = false;
+                        validation.errorMessages.push(propertyMetadata.displayName + " cannot have more than " +
```

Comment:
Can we define this error message also as an i18n String?

Reply:
An i18n String has not been used in all the places where this error message is needed (used 4 times in the file) since the maximum length for each property is different, so we cannot have the error message defined as a single constant value.